### ISSUE TITLE
CUDA not available on RTX 4080

### ISSUE THREAD

**Issue by vadi2** [pytorch/pytorch#91973]
### 🐛 Describe the bug

CUDA for some reason is not available, not even on latest source build (see below) even though all issues linked to https://github.com/pytorch/pytorch/issues/91122 are closed.

### Versions

```
Collecting environment information...
PyTorch version: 2.0.0a0+git7f2b5ea
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.1 LTS (x86_64)
GCC version: (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.8 (main, Nov 24 2022, 14:13:03) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-57-generic-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 4080
Nvidia driver version: 525.60.11
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.24.1
[pip3] torch==1.13.1
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.3.1               h2bc3f7f_2  
[conda] magma-cuda116             2.6.1                         1    pytorch
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-include               2022.1.0           h06a4308_224  
[conda] mkl-service               2.4.0           py310h7f8727e_0  
[conda] mkl_fft                   1.3.1           py310hd6ae3a3_0  
[conda] mkl_random                1.2.2           py310h00e6091_0  
[conda] numpy                     1.24.1                   pypi_0    pypi
[conda] numpy-base                1.23.5          py310h8e6c178_0  
[conda] pytorch                   1.13.1             py3.10_cpu_0    pytorch
[conda] pytorch-mutex             1.0                         cpu    pytorch
[conda] torch                     2.0.0a0+git7f2b5ea           dev_0    <develop>
```

**Comment by greenbrettmichael** [comment]
I struggled with this for 3 hours last night after I upgraded! 
For me, it was that I had an incompatible version of CUDA installed, 11.7. Support for Lovelace GPUs was added in CUDA 11.8.
I had to uninstall CUDA 11.7 and install CUDA 11.8 using the instructions on the NVIDIA site.
I currently have it working with the prebuilt pytorch 1.13 + CUDA 11.7 installed through conda.

**Comment by ptrblck** [comment]
```
pytorch                   1.13.1             py3.10_cpu_0
```
shows that you've installed the CPU-only binary.
The current PyTorch binaries with CUDA 11.7 work on your 4080 since the compute capability (sm_89) is compatible with sm_80. We are aware of one issue in `cuFFT` described [here](https://github.com/pytorch/pytorch/issues/88038), which indeed needs an updated cuFFT version from CUDA 11.8. You could install the experimental CUDA 11.8 nightly binaries if you are running into this issue and depend on cuFFT for your workloads.

**Comment by vadi2** [comment]
I've compiled from sources following https://github.com/pytorch/pytorch#from-source. I've had to install magma-cuda116 since that was the latest package though.

What would you like me to do differently?

The current binaries do not work for me btw, they say CUDA is not available - that's why I tried compiling the latest code.

**Comment by ptrblck** [comment]
Building from source is certainly an option, but as already described your environment shows that you've installed the CPU-only binaries, which do not provide CUDA support.
Use the [install instructions from here](https://pytorch.org/get-started/locally/), select the right CUDA runtime, copy/paste the install command, and execute it in your environment. The install logs show which version will be installed.
If you are still having trouble in your environment, I would recommend to create a new and empty virtual environment and install the right binaries there.

**Comment by vadi2** [comment]
CUDA 12.0 is not listed as an option in the page you link. What is "the right CUDA runtime" in this case?

**Comment by ptrblck** [comment]
The binaries with CUDA 11.6, 11.7, and the (experimental) 11.8 runtime will work as already described.

**Comment by vadi2** [comment]
Yep confirmed, it does work. The previous instructions from elsewhere I was using `conda install pytorch cudatoolkit=11.3 -c pytorch` and that was breaking things.
